helix-core/src/test/java/org/apache/helix/TestListenerCallback.java

```diff
@@ -195,19 +205,27 @@ public void testScopedConfigChangeListener() throws Exception {
 
     listener.reset();
     _manager.addConfigChangeListener(listener, ConfigScopeProperty.CLUSTER);
-    Assert.assertTrue(listener._configChanged, "Should get initial clusterConfig callback invoked");
+    boolean result = TestHelper.verify(()-> {
+      return listener._configChanged;
+    }, TestHelper.WAIT_DURATION);
+    Assert.assertTrue(result,"Should get initial clusterConfig callback invoked");
     Assert.assertEquals(listener._configSize, 1, "Cluster Config size should be 1");
 
     listener.reset();
     _manager.addConfigChangeListener(listener, ConfigScopeProperty.RESOURCE);
+    result = TestHelper.verify(()-> {
```

Comment:
Assert result

Reply:
Good catch. This is actually addressed in https://github.com/apache/helix/pull/1447/files. This file change also belongs there.